### PR TITLE
build: increase fetch-deps

### PR DESCRIPTION
**Pull request by VerteDinde**
#### Description of Change

This PR increases the fetch-depth in checkout, and fetches tags as well as commit history.
~It also re-enables SSH debugging with tmate, so it's easier for us to SSH into the Actions runner while debugging.~ Removed this functionality in this PR

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant documentation, tutorials, templates and examples are changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: none

### CODE REVIEW

File: .github/workflows/macos-build.yml [hunk 1]
```diff
@@ -210,6 +210,7 @@ jobs:
       with:
         path: src/electron
         fetch-depth: 0
+        fetch-tags: true
```
Comment (codebytere) [inline .github/workflows/macos-build.yml:213]: this is the default with fetch-depth: 0 so we can omit this i'd say!

File: .github/workflows/macos-build.yml [hunk 2]
```diff
@@ -293,6 +300,8 @@ jobs:
       uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29
       with:
         path: src/electron
+        fetch-depth: 0
+        fetch-tags: true
```
Comment (codebytere) [inline .github/workflows/macos-build.yml:304]: same as above

**Review by codebytere — APPROVED** [review]

**Review by MarshallOfSound — CHANGES_REQUESTED** [review]
We should pin this to to a sha, but also this was supposed to be short term, why do we need to add this back in to the now privileged release jobs?

**Comment by VerteDinde** [comment]
@MarshallOfSound For context: upload is failing on the publish job in the actual release, but it's difficult to see exactly why. The hope here was to be able to SSH in after a failure and more easily figure out what was happening, and then removing the action once that debugging happened. But I'm fine removing it from the `main` job, we can set up our own tmate server or debug a different way 🙂

**Review by MarshallOfSound — APPROVED** [review]

**Pull request merged**